snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint_type.groovy

```diff
@@ -240,14 +259,82 @@ def searchRelationshipsWithUnregulatedTypeIds =  {
 		owlMemberExpressionBuilder.filter(SnomedRefSetMemberIndexEntry.Expressions.effectiveTime(EffectiveTimes.UNSET_EFFECTIVE_TIME))
 	}
 	
-	final Query<String> owlMemberQuery = Query.select(String.class)
+	final Query<String[]> owlMemberQuery = Query.select(String[].class)
 		.from(SnomedRefSetMemberIndexEntry.class)
-		.fields(SnomedRefSetMemberIndexEntry.Fields.ID)
+		.fields(SnomedRefSetMemberIndexEntry.Fields.ID, SnomedRefSetMemberIndexEntry.Fields.OWL_EXPRESSION)
 		.where(owlMemberExpressionBuilder.build())
 		.limit(50_000)
 		.build();
 	
-	searcher.stream(owlMemberQuery).forEach({ hits -> });
+	searcher.stream(owlMemberQuery).forEach({ hits ->
+		hits.each { hit ->
+			def id = hit[0];
+			def owlExpression = hit[1];
+			def hasAxiomRelationships = !unsupportedAxiomMarkers.stream().filter({ marker -> owlExpression.contains(marker)}).findAny();
```

Comment:
Hmm. It might be easier to check whether the expression starts with `SubClassOf` or `EquivalentClasses` 😃

Reply:
My comment doesn't hold true for the object and data property hierarchies, see section "Representation of |is a| Relationships" in the SNOMED CT OWL Guide: https://confluence.ihtsdotools.org/display/DOCOWL/2.4.+Content+for+the+OWL+Axiom+Refset

Only property roots receive a `SubClassOf` axiom that connects them to the `410662002|Concept model attribute (attribute)|`; the rest of each hierarchy is modeled using `SubObjectPropertyOf` and `SubDataPropertyOf` axioms, respectively.